src/plugins/vis_type_timeseries/public/application/components/aggs/field_select.tsx

```diff
@@ -128,7 +128,7 @@ export function FieldSelect({
       selectedOptions = [{ label: value!, id: 'INVALID_FIELD' }];
     }
   } else {
-    if (value && !selectedOptions.length) {
+    if (value && fields[fieldsSelector] && !selectedOptions.length) {
       onChange([]);
```

Comment:
I would prefer to remove that `onChange` callback cause it's an example of bad code from the `React` perspective. But it was implemented historically and solve some business needs....

Reply:
Yes I agree Alex, do you remember which business needs solves? Maybe we could add a comment to remind us why this code exists.